packages/@lwc/style-compiler/src/transform.ts

```diff
@@ -30,7 +30,7 @@ export function transform(src: string, id: string, config: Config = {}): { code:
         return { code: 'export default undefined' };
     }
 
-    const allowDefinition = !config.customProperties || config.customProperties.allowDefinition;
+    const allowDefinition = config.customProperties?.allowDefinition ?? false;
```

Comment:
will this change behavior for when `customProperties` is not defined?

```js
const config = {};

const oldWay = !config.customProperties || config.customProperties.allowDefinition;
const newWay = config.customProperties?.allowDefinition ?? false;

console.log('old', oldWay);
console.log('new', newWay);
```

Reply:
Good catch let's keep it like that for now. The current implementation doesn't match the documentation. The default value should be `false`. Let me revert this for now, I will fix this in another PR.